### PR TITLE
Add support for the DMA widget

### PR DESCRIPTION
**Pull request by ArthurHeymans**

### CODE REVIEW

**Review by swenson — APPROVED** [review]
LGTM. Would be nice to get some clarity on those TODOs, but this looks good for now to unblock us.

**Review by mhatrevi — CHANGES_REQUESTED** [review]

File: sw-emulator/app/src/main.rs [hunk 1]
```diff
@@ -82,6 +82,11 @@ fn main() -> io::Result<()> {
             arg!(--"rom" <FILE> "ROM binary path")
                 .value_parser(value_parser!(PathBuf))
         )
+        .arg(
```
[the remark this answers is not in the document]
Comment (mhatrevi) [reply sw-emulator/app/src/main.rs:85]: Can do something simple like recovery-image-fw / recovery-image-manifest, recovery-image-mcurt as param names.

File: sw-emulator/app/src/main.rs [hunk 2]
```diff
@@ -155,6 +170,9 @@ fn main() -> io::Result<()> {
         .get_matches();
 
     let args_rom = args.get_one::<PathBuf>("rom").unwrap();
+    let args_recovery_fw = args.get_one::<PathBuf>("recovery-image-fw");
+    let _args_recovery_image_manifest= args.get_one::<PathBuf>("recovery-image-manifest"); // TODO hook up to RRI
+    let _args_recovery_image_mcurt= args.get_one::<PathBuf>("recovery-image-mcurt"); // TODO hook up to RRI
     let args_current_fw = args.get_one::<PathBuf>("firmware");
```
Comment (mhatrevi) [inline sw-emulator/app/src/main.rs:176]: let args_current_fw = args.get_one::<PathBuf>("firmware"); 

This should be removed as recovery-image-fw is the firmware. Also, rename the update-firmware to recover-image-fw-update

**Review by mhatrevi — APPROVED** [review]

**Review by swenson — APPROVED** [review]
LGTM

**Pull request merged**